File: leet_easy_c++/leet_110/leet_110.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isBalanced(TreeNode* root) {
        if(!root) return true;
        if(tree_depth(root->left) > tree_depth(root->right) + 1 || tree_depth(root->left) + 1 < tree_depth(root->right)) 
            return false;
        else{
            return isBalanced(root->right) && isBalanced(root->left);
        }
    }
    
    int tree_depth(TreeNode* cur){
        if(cur == NULL) return 0;
        if(cur->left == NULL && cur->right == NULL) return 1;
        if(cur->right != NULL && cur->left != NULL) return max(tree_depth(cur->left), tree_depth(cur->right)) + 1;
        if(cur->right != NULL) return tree_depth(cur->right) + 1;
        if(cur->left != NULL) return tree_depth(cur->left) + 1;
    }
};

File: leet_easy_python/leet_110/leet_110.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isBalanced(self, root):
        self.flag = True
        self.iBh(root)
        return self.flag
        """
        :type root: TreeNode
        :rtype: bool
        """
        
    def iBh(self, iter): #iBh = isBalanced helper
        if not iter or self.flag == False: return 0
        l = self.iBh(iter.left) + 1
        r = self.iBh(iter.right) + 1
        if abs(l-r) > 1: self.flag = False
        return max(l,r)
        """
        :type iter: TreeNode
        :rtype: int
        """